.github/workflows/README.md

```diff
@@ -29,13 +29,14 @@ code should be bumped to `X.Y.(Z+1)-dev0`.
 For each release the code will be tagged with `v<version number>`.
 
 ### Pre-releases
-A pre-release does not have public release page on GitHub, but is uploaded to pypi 
-(however, it will only be installed if the `--pre` flag is passed to `pip`, i.e. `pip install --pre <packagename>`).
-This mechanism is meant to speed up the internal development.
+A pre-release is uploaded to pypi, but it will only be installed if the version number is explicitly given 
+or if the `--pre` flag is passed to `pip` (i.e. `pip install --pre <packagename>`).
+This way, the pre-releases are not pulled by dependent packages, unless explicitly requested.
+We use this mechanism to speed up the development.
 If a pre-release of `X.Y.Z-dev0` is done, the next version number should be `X.Y.Z-dev1`.
 
 (Note: if you want to build wheel from such a version, you need to substitute `-dev` with `.dev`,
-to have e.g. `packaname-1.2.3.dev0-py3-none-any.whl`)
+to have e.g. `packagename-1.2.3.dev0-py3-none-any.whl`)
 
 
 ### Step-by-step instructions
```

Comment:
Maybe add what we will do on a high level:

bump the version away from a dev version,
perform release,
Switch to dev